packages/pegasus/src/ics20.js

```diff
@@ -0,0 +1,102 @@
+// @ts-check
+import { Nat } from '@agoric/nat';
+import { Far } from '@endo/marshal';
+import { assert, details as X } from '@agoric/assert';
+
+/**
+ * @typedef {Object} ICS20TransferPacket
+ * @property {string} amount The extent of the amount
+ * @property {Denom} denom The denomination of the amount
+ * @property {string} [sender] The sender address
+ * @property {DepositAddress} receiver The receiver deposit address
+ */
+
+/**
+ * Convert an inbound packet to a local amount.
+ *
+ * @param {Bytes} packet
+ * @returns {Promise<PacketParts>}
+ */
+export const parseICS20TransferPacket = async packet => {
+  /** @type {ICS20TransferPacket} */
```

Comment:
I don't think we've established this type just by doing a JSON.parse.

```suggestion
  /** @type {unknown} */
```